src/llm/apis/openai_completions.cpp

```diff
@@ -75,32 +76,139 @@ absl::Status OpenAIChatCompletionsHandler::parseCompletionsPart() {
             return absl::InvalidArgumentError("Message is not a JSON object");
-        auto& chat = request.messages.emplace_back(chat_entry_t{});
         for (auto member = obj.MemberBegin(); member != obj.MemberEnd(); member++) {
             if (!member->name.IsString())
                 return absl::InvalidArgumentError("Invalid message structure");
-            if (!member->value.IsString())
-                return absl::InvalidArgumentError("Invalid message structure");
-            chat[member->name.GetString()] = member->value.GetString();
+            if (member->value.IsString()) {
+                continue;
+            } else {
+                if (member->name.GetString() == std::string("content") && member->value.IsArray()) {
+                    jsonChanged = true;
+                    Value contentText;
+                    for (auto& v : member->value.GetArray()) {
+                        if (!v.IsObject()) {
+                            return absl::InvalidArgumentError("Invalid message structure - content array should contains objects");
+                        }
+                        auto entry = v.GetObject();
+                        if (!entry.HasMember("type") || !entry["type"].IsString()) {
+                            return absl::InvalidArgumentError("Invalid message structure - content object type missing");
+                        }
+                        auto type = entry["type"].GetString();
+                        if (type == std::string("text")) {
+                            if (!entry.HasMember("text") || !entry["text"].IsString()) {
+                                return absl::InvalidArgumentError("Invalid message structure - content text missing");
+                            }
+                            contentText = entry["text"];
+                            continue;
+                        } else if (type == std::string("image_url")) {
+                            if (!entry.HasMember("image_url") || !entry["image_url"].IsObject()) {
+                                return absl::InvalidArgumentError("Invalid message structure - content image_url missing");
+                            }
+                            auto imageUrl = entry["image_url"].GetObject();
+                            if (!imageUrl.HasMember("url") || !imageUrl["url"].IsString()) {
+                                return absl::InvalidArgumentError("Invalid message structure - image_url does not have url field");
+                            }
+                            std::string url = imageUrl["url"].GetString();
+                            std::string pattern = ";base64,";
+                            std::size_t found = url.find(pattern);
+                            size_t offset = found + pattern.length();
+                            if (found == std::string::npos) {
+                                offset = 0;
+                            }
+                            url.erase(0, offset);
```

Comment:
do we need to erase? maybe we could construct the vector by passing begin offseted by patternPosition value?